go/kbfs/libkbfs/favorites.go

```diff
@@ -336,6 +348,10 @@ func (f *Favorites) handleReq(req *favReq) (err error) {
 			}
 			// If we weren't explicitly asked to refresh, we can return possibly
 			// stale favorites rather than return nothing.
+			if err == context.DeadlineExceeded {
+				// TODO: make sure this doesn't get queued like 10 times
+				go f.RefreshCache(context.Background())
```

Comment:
This should probably use a real context with a real timeout, derived from `req.ctx` preferably so log tags are preserved.

Reply:
If I use one derived from `req.ctx`, then it will get cancelled if that gets cancelled, which would be unfortunate if it were to happen soon after.

Reply:
Oh, oops, apparently internally we mostly ignore this context anyway:
```

// RefreshCache refreshes the cached list of favorites.
func (f *Favorites) RefreshCache(ctx context.Context) {
	if f.disabled || f.hasShutdown() {
		return
	}
	// This request is non-blocking, so use a throw-away done channel
	// and context.
	req := &favReq{
		refresh: true,
		done:    make(chan struct{}),
		ctx:     context.Background(),
	}
	f.wg.Add(1)
	select {
	case f.reqChan <- req:
	case <-ctx.Done():
		f.wg.Done()
		return
	}
}
```